File: dices.jl
#!/usr/bin/env julia

open("input.txt") do file
    sum = 0
    for ln in eachline(file)
        rolls, faces = split(ln, "d")
        rolls = parse(Int32, rolls)
        faces = parse(Int32, faces)
        for i = 1:rolls
            sum += rand(1:faces)
        end
    end
    println("$(sum)")
end

File: dices.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main()
{
    int dados, faces ,sum, i;
    FILE *fp = fopen("input.txt","r+");
    while(!feof(fp)) {
        fscanf(fp,"%dd%d\n",&dados,&faces);
        sum=0;
        for(i = 0; i < dados; i++)
            sum+=rand() % faces + 1;
        printf("%d\n",sum);
    }
    fclose(fp);
    return 0;
}
